streamdeck_ui/api.py

```diff
@@ -699,3 +751,41 @@ def _update_button_filters(self, serial_number: str, page: int, button: int):
             )
 
         display_handler.replace(page, button, filters)
+
+    def prepare_plugin(self, serial_number: str, page_id: int, button_id: int, plugin_path: str) -> Union[Plugin, None]:
```

Comment:
What do you think about prepare_plugin to be decouple from API. instead using self inside can receive as parameters the config and a interface with all the supported callbacks. that allow us to test easily the logic about preparing the plugin, leaving API with simple calls

Reply:
That makes sense, I could put it back into plugins.py where it was originally and make self be replaced with an interface containing the api callbacks or the references to them. This way the api doesnt even need to be passed in directly.
Im not 100% sure how I can make that happen but it makes more sense and is better for testing and a more centralized logic like you said

Reply:
start by inverting the dependencies (like API) and then passing just plain objects instead classes instances, so you can mock easily each parameter without the entire thing